### PR TITLE
*: format code

### PR DESCRIPTION
**Pull request by follitude**
This pr updates rustfmt to rustfmt-nightly 0.2.1.
PTAL @siddontang @zhangjinpeng1987

### CODE REVIEW

**Comment by siddontang** [comment]
CI failed

**Comment by siddontang** [comment]
CI still failed, seem we can only check format in nightly version

**Comment by follitude** [comment]
fixed @siddontang

**Comment by siddontang** [comment]
LGTM

PTAL @BusyJay

File: librocksdb_sys/crocksdb/c.cc [hunk 1]
```diff
@@ -1644,12 +1644,12 @@ crocksdb_compactionjobinfo_table_properties(
 }
 
 uint64_t crocksdb_compactionjobinfo_elapsed_micros(
-    const crocksdb_compactionjobinfo_t* info) {
+    const crocksdb_compactionjobinfo_t *info) {
```
Comment (BusyJay) [inline librocksdb_sys/crocksdb/c.cc]: I don't think this is the expected format.

File: .travis.yml [hunk 2]
```diff
@@ -1,29 +1,37 @@
...
+  include:
+    - os: linux
+      rust: nightly
+      env: SKIP_FORMAT_CHECK=false
+    - os: linux
+      rust: beta
+      env: SKIP_FORMAT_CHECK=true
+    - os: linux
+      rust: stable
+      env: SKIP_FORMAT_CHECK=true
```
Comment (BusyJay) [inline .travis.yml]: I think you can disable format check in all travis ci, since circle ci has checked it already.

File: .clang-format [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Complete list of style options can be found at: 
+# http://clang.llvm.org/docs/ClangFormatStyleOptions.html
+---
+BasedOnStyle: Google
```
Comment (siddontang) [inline .clang-format]: seem we must follower RocksDB style, not Google.

Comment (andelf) [inline .clang-format]: It was once mentioned.
We've borrowed all format facilities from facebook/rocksdb, but forgot to copy the `.clang-format` configuration file. This is the fix.

ref: https://github.com/facebook/rocksdb/blob/master/.clang-format

Comment (siddontang) [inline .clang-format]: got it

**Comment by siddontang** [comment]
LGTM

PTAL @BusyJay

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**